registry/Scripts/pre-reqs.ps1

```diff
@@ -65,6 +68,173 @@ function New-ResourceGroup (
+#>
+function New-DnsZone (
+    [Parameter(Mandatory = $true, HelpMessage = "DNS zone's resource group.")]
+    [string] $ResourceGroupName,
+    [Parameter(Mandatory = $true, HelpMessage = "DNS zone name.")]
+    [string] $Name
+)
+{
+    # Create storage account
+    
+    Write-Host "Check if DNS Zone ($Name) already exists"
+    $ErrorActionPreference = "SilentlyContinue";
+    $dnsZoneDetails = Get-AzureRmDnsZone -Name $Name `
+                                         -ResourceGroupName $ResourceGroupName `
+                                         -ErrorVariable dnsZoneExistError
+    $ErrorActionPreference = "Continue"; #Turning errors back on
+    if (-not $dnsZoneDetails) {
+        Write-Host "DNS zone does not exists. Creating a new DNS zone ($Name) under resource group ($ResourceGroupName)." 
+        New-AzureRmDnsZone -Name $Name -ResourceGroupName $ResourceGroupName | Out-Null
+        #$ErrorActionPreference = "SilentlyContinue";
+        $newDNSZoneDetails = Get-AzureRmDnsZone -Name $Name -ResourceGroupName $ResourceGroupName `
+                                                       -ErrorVariable newDNSZoneCreateError
+        #$ErrorActionPreference = "Continue"; #Turning errors back on
+        if  (-not $newDNSZoneDetails) {
+            throw "Creation a new DNS zone ($Name) under resource group ($ResourceGroupName) failed. Check if specified DNS zone already exists under a subscription." 
+        }
+        Write-Host "DNS zone ($Name) created successfully." 
+    }
+    else {        
+        Write-Host "DNS zone ($Name) already exists. Skipping creation."
+    }
+
+    return $storageAccountDetails
+}
+
+<#
+.Synopsis
+    Create/update DNS record under given DNS zone.
+
+.Description
+    Create/update DNS record under given DNS zone.
+    It will update DNS record if already present.
+
+.Parameter ResourceGroupName
+    DNS zone's resource group.
+
+.Parameter ZoneName
+    DNS zone name.
+
+.Parameter Name
+    DNS record name.
+
+.Parameter RecordType
+    Type of record.
+
+.Parameter DnsRecordConfigValue
+    Record value.
+
+.Parameter Ttl
+    Ttl value.
+
+.Example
+   Set-DnsRecordSet -ResourceGroupName "resourcegroupname"
+                    -ZoneName "sample.com"
+                    -Name "test"
+                    -RecordType "A"
+                    -DnsRecordConfigValue "0.0.0.0"
+#>
+function Set-DnsRecordSet (
+    [Parameter(Mandatory = $true, HelpMessage = "DNS zone's resource group.")]
+    [string] $ResourceGroupName,
+    [Parameter(Mandatory = $true, HelpMessage = "DNS zone name.")]
+    [string] $ZoneName,
+    [Parameter(Mandatory = $true, HelpMessage = "DNS record name.")]
+    [string] $Name,
+    [Parameter(Mandatory = $true, HelpMessage = "Type of record.")]
+    [string] $RecordType,
+    [Parameter(Mandatory = $true, HelpMessage = "Record value.")]
+    [string] $DnsRecordConfigValue,
+    [Parameter(Mandatory = $false, HelpMessage = "Ttl value.")]
```

Comment:
`TTL`

Reply:
I used to align the way commandlet is taking input. But will update.